pkg/secrets-store/nodeserver.go

```diff
@@ -60,10 +60,17 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	var parameters map[string]string
 	var providerName string
 	var podName, podNamespace, podUID string
+	var targetPath string
 	errorReason := FailedToMount
 
 	defer func() {
 		if err != nil {
+			// if there is an error at any stage during node publish volume, we need
+			// to unmount the target path so that the next time kubelet calls again for
+			// mount, we retry the whole node publish volume
+			if targetPath != "" {
+				ns.mounter.Unmount(targetPath)
```

Comment:
should we first check if the path is mounted?

Reply:
We should add a test for this.

Reply:
Added `mounted` boolean we can leverage before deciding if we need to unmount. The e2e test scenario would require the provider to respond only after the context deadline for this particular case.

Reply:
How about we add an e2e test for this to a long running periodic job?